packages/zwave-js/src/lib/commandclass/SceneActuatorConfigurationCC.ts

```diff
@@ -0,0 +1,395 @@
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	): Promise<void> => {
+		if (propertyKey == undefined) {
+			throwMissingPropertyKey(this.ccId, property);
+		} else if (typeof propertyKey !== "number") {
+			throwUnsupportedPropertyKey(this.ccId, property, propertyKey);
+		}
+		if (property === "level") {
+			if (typeof value !== "number") {
+				throwWrongValueType(
+					this.ccId,
+					property,
+					"number",
+					typeof value,
+				);
+			}
+
+			// We need to set the dimming duration along with the level
+			const node = this.endpoint.getNodeUnsafe()!;
+			// If duration is missing, we set a default of instant
+			const dimmingDuration =
+				node.getValue<Duration>(
+					getDimmingDurationValueID(this.endpoint.index, propertyKey),
+				) ?? new Duration(0, "seconds");
+			await this.set(propertyKey, dimmingDuration);
+		} else {
+			// setting dimmingDuration value alone not supported,
+			// to be added when setting Duration values is supported
+			throwUnsupportedProperty(this.ccId, property);
+		}
+
+		// Verify the current value after a delay
+		this.schedulePoll({ property, propertyKey });
+	};
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+		propertyKey,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "level":
+			case "dimmingDuration": {
+				if (propertyKey == undefined) {
+					throwMissingPropertyKey(this.ccId, property);
+				} else if (typeof propertyKey !== "number") {
+					throwUnsupportedPropertyKey(
+						this.ccId,
+						property,
+						propertyKey,
+					);
+				}
+				return (await this.get(propertyKey))?.[property];
+			}
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	public async set(
+		sceneId: number,
+		dimmingDuration: Duration,
+		level?: number,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			SceneActuatorConfigurationCommand,
+			SceneActuatorConfigurationCommand.Set,
+		);
+
+		const cc = new SceneActuatorConfigurationCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			sceneId,
+			dimmingDuration,
+			level,
+		});
+
+		await this.driver.sendCommand(cc, this.commandOptions);
+	}
+
+	public async get(
+		sceneId: number,
+	): Promise<
+		| {
+				sceneId: number;
+				level: number;
+				dimmingDuration: Duration;
+		  }
+		| undefined
+	> {
+		this.assertSupportsCommand(
+			SceneActuatorConfigurationCommand,
+			SceneActuatorConfigurationCommand.Get,
+		);
+
+		const cc = new SceneActuatorConfigurationCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			sceneId: sceneId,
+		});
+		const response = await this.driver.sendCommand<SceneActuatorConfigurationCCReport>(
+			cc,
+			this.commandOptions,
+		);
+
+		if (response) {
+			return pick(response, ["sceneId", "level", "dimmingDuration"]);
```

Comment:
If 0 wasn't used to retrieve the current scene, I'd remove `sceneId` from the return value.
I already didn't like this in Scene Controller Configuration CC, maybe we should split the API command into two:
`getCurrentScene()` and `get(sceneId)` with `sceneId > 0`?

Reply:
We definitely should have `getCurrentScene()` in the API.

If we want to require `sceneId > 0` for `get(sceneId)` (and omit `sceneId` from the return value), we should update Scene Controller Configuration CC to behave the same way.  That would be a breaking change, but I doubt it would mess anyone up, since it was just added.

Reply:
Agreed.

>  That would be a breaking change, but I doubt it would mess anyone up, since it was just added.

Also agreed!